backend/ecom/routes/address.route.js

```diff
@@ -0,0 +1,33 @@
+const express = require('express');
+const router = express.Router();
+const Address = require('../model/address.model');
+const { extend } = require('lodash');
+
+router.route('/')
+  .get(async (req, res) => {
+    try {
+      const user = req.user;
+      const userAddresses = await Address.findOne({ _id: user.addressId }).populate('address');
+      res.json({ success: true, userAddresses });
+    }
+    catch (error) {
+      res.status(401).json({ success: false, message: "error in catch block" })
+    }
+
+  })
+  .post(async (req, res) => {
+    try {
+      const user = req.user;
+      const userAddresses = await Address.findOne({ _id: user.addressId });
+      userAddresses.address.push({ ...req.body });
+      await userAddresses.save();
+      const updatedUserAddresses = await Address.findOne({ _id: user.addressId }).populate('address');
```

Comment:
You don't need to make another DB call to fetch the updated addresses of a user. After `line 22` the `userAddresses.address` will have the pushed address which you can simply send in the response and yes it'll contain the `_id` field and everything a saved address object would've. You may `console.log()` the output if you don't believe me ☻

```suggestion
res.json({ success: true, updatedUserAddresses: userAddresses.address });
```